lumicks/pylake/correlated_stack.py

```diff
@@ -14,13 +19,103 @@ class TiffFrame:
+            # takes into account both the offset and the scaling performed by the first step
+            transform3 = np.eye(3)
+            transform3[0, -1] = transform2[0, 0]*x_offset
+            transform3[1, -1] = transform2[1, 1]*y_offset
+            # concatenate the transforms by multiplying their matrices and ignore the unused 3rd row
+            return np.dot(transform3, np.dot(transform2, transform1))[:2, :]
+
+        description = json.loads(self._src.description)
+        try:
+            align_mats = [np.array(description[f"Alignment {color} channel"]).reshape((2,3))
+                        for color in ('red', 'blue')]
+            align_roi = np.array(description["Alignment region of interest (x, y, width, height)"])[:2]
+            roi = np.array(description["Region of interest (x, y, width, height)"])[:2]
+        except KeyError:
+            warnings.warn("File does not contain alignment matrices. Only raw data is available")
+            return self.raw_data
+        
+        x_offset, y_offset = align_roi - roi
+        if not (x_offset == 0 and y_offset == 0):
+            align_mats = [correct_alignment_offset(mat, x_offset, y_offset) for mat in align_mats]
+
+        img = self.raw_data
+        rows, cols, _ = img.shape
+        for mat, channel in zip(align_mats, (0, 2)):
+            img[:,:,channel] = cv2.warpAffine(img[:,:,channel], mat, (cols,rows),
+                                              flags=(cv2.INTER_LINEAR | cv2.WARP_INVERSE_MAP),
+                                              borderMode=cv2.BORDER_CONSTANT, borderValue=0)
+        return img
+        
     @property
     def data(self):
+        return self._align_image() if (self.is_rgb and self._align) else self._src.asarray()
+
+    @property
+    def raw_data(self):
         return self._src.asarray()
 
+    @property
+    def bit_depth(self):        
+        bit_depth = self._src.tags["BitsPerSample"].value
+        try:
+            return bit_depth[0]
+        except TypeError:
+            return bit_depth
+
+    @property
+    def is_rgb(self):
+        return self._src.tags["SamplesPerPixel"].value == 3
+
+    def _get_plot_data(self, channel="rgb", vmax=None): 
+        """ return data an numpy array, convert dtype=float in range [0,1] 
+            if data is grayscale or channel in ('red', 'green', 'blue')
+                return data as is
+            if channel is 'rgb', correct for optional vmax argument:
+                None   : return data as is
+                'auto' : normalize data to max signal of all channels
+                float  : normalize data to vmax value
+        """
+        data = (self.data/(2**self.bit_depth-1)).astype(np.float)
+        
+        if not self.is_rgb:
+            return data
+
+        if (channel is None) or (channel.lower() == "rgb"):
+            if vmax is None:
+                return data
+            elif isinstance(vmax, (int, float)):
+                return data/vmax
```

Comment:
This approach works, but ends up showing a ton of warnings from imshow which mention that the data is clamped. Would be nice to handle that more gracefully.